torch/_inductor/codegen/cpp_micro_gemm.py

```diff
@@ -504,18 +504,67 @@ class CppMicroGemmAMX(CppMicroGemm):
     This class generates the code for micro gemm using Advanced Matrix eXtention (AMX)
     instructions available in 4th generation Intel Xeon for compute.
     It supports input types of torch.bfloat16 with fp32 output.
-    TODO(jgong5): support int8 data type.
     """
 
     TEMPLATE_ENTRY = r"""
 {{declare_kernel}} {
     {{kernel.assert_function}}(N % {{block_n}} == 0, "N dimension must be multiple of {{block_n}}");
     {{kernel.assert_function}}(K % 2 == 0, "K dimension must be multiple of 2");
+{%- if input_dtype == torch.bfloat16 and input2_dtype == torch.int8 %}
```

Comment:
Instead of checking this condition everywhere, we may consider to wrap it with some flag with more meaningful, like `need_dequantize_b` (you can give a better name)?

Reply:
Perhaps we can factor out the dequantization of B into a codegen function so that it can be placed in any place where proper. Not necessary in this PR though.

Reply:
> Perhaps we can factor out the dequantization of B into a codegen function so that it can be placed in any place where proper. Not necessary in this PR though.

If dequantization of B would be moved out of `TEMPLATE_ENTRY`, then it'd conflict with similar declarations pertaining to other AMX kernels.

Reply:
> > Perhaps we can factor out the dequantization of B into a codegen function so that it can be placed in any place where proper. Not necessary in this PR though.
> 
> If dequantization of B would be moved out of `TEMPLATE_ENTRY`, then it'd conflict with similar declarations pertaining to other AMX micro-kernels.

I mean we can factor out a codegen function like `dequantize_buffer` to be called inside other templates. Why would that create conflict?

Reply:
Giving it a name like `sequantize_buffer` would result in it being duplicated, so I could name it the same way AMX micro-kernel is being named (with num_rows & `num_columns` in the name).

Reply:
Thanks, will do that in another PR, since with int4, it'd be better to have a separate codegen function for dequantizing weights.